workitem/event/event_repository.go

```diff
@@ -0,0 +1,198 @@
+package event
+
+import (
+	"context"
+	"strings"
+
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/fabric8-services/fabric8-wit/account"
+	"github.com/fabric8-services/fabric8-wit/workitem"
+)
+
+const (
+
+	// APIStringTypeEvents is the name of event type
+	APIStringTypeEvents = "events"
+
+	Assignees = "assignees"
+	State     = "state"
+	Labels    = "labels"
+	Iteration = "iteration"
+)
+
+// WorkItemEventRepository encapsulates retrieval of work item events
+type WorkItemEventRepository interface {
+	//repository.Exister
+	List(ctx context.Context, wiID uuid.UUID) ([]WorkItemEvent, error)
+}
+
+// NewWorkItemEventRepository creates a work item event repository based on gorm
+func NewWorkItemEventRepository(db *gorm.DB) *GormWorkItemEventRepository {
+	return &GormWorkItemEventRepository{
+		db:               db,
+		workItemRepo:     workitem.NewWorkItemRepository(db),
+		wiRevisionRepo:   workitem.NewRevisionRepository(db),
+		workItemTypeRepo: workitem.NewWorkItemTypeRepository(db),
+		identityRepo:     account.NewIdentityRepository(db),
+	}
+}
+
+// GormWorkItemEventRepository represents the Gorm model
+type GormWorkItemEventRepository struct {
+	db               *gorm.DB
+	workItemRepo     *workitem.GormWorkItemRepository
+	wiRevisionRepo   *workitem.GormRevisionRepository
+	workItemTypeRepo *workitem.GormWorkItemTypeRepository
+	identityRepo     *account.GormIdentityRepository
+}
+
+// List return the events
+func (r *GormWorkItemEventRepository) List(ctx context.Context, wiID uuid.UUID) ([]WorkItemEvent, error) {
+	revisionList, err := r.wiRevisionRepo.List(ctx, wiID)
+	if err != nil {
+		return nil, errs.Wrapf(err, "error during fetching event list")
+	}
+	if revisionList == nil {
+		return []WorkItemEvent{}, nil
+	}
+	wi, err := r.workItemRepo.LoadByID(ctx, wiID)
+	if err != nil {
+		return nil, errs.Wrapf(err, "error during fetching event list")
+	}
+	wiType, err := r.workItemTypeRepo.LoadByID(ctx, wi.Type)
+	if err != nil {
+		return nil, errs.Wrapf(err, "error during fetching event list")
+	}
+
+	eventList := []WorkItemEvent{}
```

Comment:
The 'event's list doesn't match the scope of the revision. You're currently creating 1 event pr change in a rev. The PATCH workitems API support more than 1 change per revision.

Checking with UX if this has been considered in the designs: https://chat.openshift.io/developers/pl/cb74c978hfg9ib8rxxkjf794wo

On the same note, the result in external API the way this is would be a []Events where multiple have the same ID (rev id) but contain different data.

Reply:
Multiple events per 1 revision is supported now: 21eb7ed627749168af3f8fb7685e48f335860bb9
Please confirm.

Reply:
Since IDs are re-used, it is not unique. This is incompatible with JSONAPI. Can I change the ID to include the field name also. So, the ID becomes UUID+fieldname. Example: `72c2ca84-2c6f-4baa-93ba-ccfa97a1c852+system.title`